# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/components/newtab/docs/v2-system-addon/about_home_startup_cache.md
# Commit: 626dd2a045af
# Full Hash: 626dd2a045af6cd510ef04e67d027909509547f1
# Author: Mike Conley <mconley@mozilla.com>
# Date: 2020-04-17 21:46:50
# Regressor Bug: 1614502
# File Overlap Count: 2
# Description:
#   Bug 1614502 - Add a mechanism that can generate an about:home document for a startup cache. r=Gijs,k88hudson
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68094
# ==============================================================================

diff -r 7ec776d31a2c -r 626dd2a045af browser/components/newtab/docs/v2-system-addon/about_home_startup_cache.md
--- a/browser/components/newtab/docs/v2-system-addon/about_home_startup_cache.md	Fri Apr 17 15:31:11 2020 +0000
+++ b/browser/components/newtab/docs/v2-system-addon/about_home_startup_cache.md	Fri Apr 17 15:31:39 2020 +0000
@@ -48,7 +48,7 @@
 
 If, at this point, nothing has been streamed from the parent, we fall back to loading the dynamic `about:home` document. This might occur if the cache doesn't exist yet, or if we were too slow to pull it off of the disk.
 
-The `AboutHomeStartupCacheChild` will also be responsible for updating the cache over time. [Bug 1614502](https://bugzilla.mozilla.org/show_bug.cgi?id=1614502) tracks that work, and this documentation will be updated once it lands with details on how the cached documents are generated.
+The `AboutHomeStartupCacheChild` will also be responsible for updating the cache over time. The `AboutHomeStartupCacheChild` will periodically request the most up-to-date state for `about:home` from the parent process, and then generate document markup using ReactDOMServer within a `ChromeWorker`. After that's generated, the "privileged about content process" will send up `nsIInputStream` instances for both the markup and the script for the initial page state. The `AboutHomeStartupCache` singleton inside of `BrowserGlue` is responsible for receiving those `nsIInputStream`'s and persisting them in the HTTP cache for the next start.
 
 ## What is cached?
 